packages/dds/tree/src/feature-libraries/chunked-forest/uniformChunk.ts

```diff
@@ -45,9 +49,21 @@ export class UniformChunk extends ReferenceCountedBase implements TreeChunk {
 	 */
 	public constructor(
 		public shape: ChunkShape,
-		public values: TreeValue[],
+		values: TreeValue[],
+		idCompressor?: IIdCompressor,
 	) {
 		super();
+		this.idCompressor = idCompressor;
+		this.values =
+			this.idCompressor !== undefined
+				? encodeUniformChunkValues(
+						values,
+						shape.treeShape,
+						shape.topLevelLength,
+						0,
+						this.idCompressor,
+					).encodedValues
+				: values;
```

Comment:
I don't think this is the ideal place to apply this logic. I think it would make more sense to do it in uniformChunkFromCursor. Specifically it (and also `insertValues` which it calls) should take in an optional idCompressor, document that string values will be compressed when possible if its provided.

This approach would avoid a second pass through the data, as well as avoid constructing the uniform chunk having to modify or copy its data array.

This would also make it possible in the future when we have optimizations for fast access to compressed identifiers for uniformChunkFromCursor to not decompress then recompress identifiers.

Also more generally, uniformChunkFromCursor will eventually get optimizations (like the above one, but also others) to allow it to process already compressed data without having to traverse it again, so integrating the logic in there will be necessary once we try and get that working efficiently and make 0 passes through the data in that special case.

Reply:
Logically we can think of UniformChunk as a tool for reading (and eventually editing) compressed data, but it doesn't do the compression itself.

Reply:
Updated to handle the encoding in the `uniformChunkFromCursor` layer